ksqldb-engine/src/main/java/io/confluent/ksql/internal/StorageUtilizationMetricsReporter.java

```diff
@@ -49,38 +48,30 @@
       = LoggerFactory.getLogger(StorageUtilizationMetricsReporter.class);
   private static final String METRIC_GROUP = "ksqldb_utilization";
 
-  private final Map<String, Map<String, TaskStorageMetric>> metricsSeen;
-  private final Metrics metricRegistry;
-  static AtomicBoolean registeredNodeMetrics = new AtomicBoolean(false);
+  private Map<String, Map<String, TaskStorageMetric>> metricsSeen;
+  private Metrics metricRegistry;
   private static Map<String, String> customTags = new HashMap<>();
 
   public StorageUtilizationMetricsReporter() {
-    this(MetricCollectors.getMetrics());
-  }
-
-  @SuppressFBWarnings("EI_EXPOSE_REP2")
-  @VisibleForTesting
-  public StorageUtilizationMetricsReporter(final Metrics metricRegistry) {
-    this.metricsSeen = new HashMap<>();
-    this.metricRegistry = Objects.requireNonNull(metricRegistry);
   }
 
   @Override
   public void init(final List<KafkaMetric> list) {
   }
 
   @Override
-  public void configure(final Map<String, ?> map) {
+  public synchronized void configure(final Map<String, ?> map) {
```

Comment:
Added synchronized to resolve an "inconsistent synchronization" warning. I'm pretty sure it's not necessary, but it's also not harmful.